src/cluster_asm.c

```diff
@@ -956,7 +1047,7 @@ void clusterSyncSlotsCommand(client *c) {
 
         /* Add main channel client to the list of slaves */
         client *main_ch_client = task->main_channel_client;
-        main_ch_client->flags |= CLIENT_SLAVE;
+        main_ch_client->flags |= (CLIENT_SLAVE | CLIENT_REPL_MIGRATION_DEST);
```

Comment:
for your approach, maybe we don't mark this client as slave, so can revert my several changes, let me try it later

Reply:
yes, I initially did it that way. Either approach is possible but not sure which one is better. In one sense, this is a slave and makes sense to mark as SLAVE. Though, it causes some problems when you don't want to forward from replication backlog directly and had to mitigate this way.

Reply:
OK, i see, we can decide later